Dockerfile

```diff
@@ -1,11 +1,14 @@
-FROM usdotfhwastoldev/carma-base:develop as base_image
+ARG DOCKER_ORG=usdotfhwastoldev
+ARG DOCKER_TAG=develop
+FROM ${DOCKER_ORG}/carma-base:${DOCKER_TAG} as base_image
 
 FROM base_image as build
+ARG GIT_BRANCH=develop
 
 COPY --chown=carma . /home/carma/autoware.auto
 RUN chmod -R 775 /home/carma/autoware.auto/docker/checkout.bash
 RUN chmod -R 775 /home/carma/autoware.auto/docker/install.sh
-RUN /home/carma/autoware.auto/docker/checkout.bash
+RUN /home/carma/autoware.auto/docker/checkout.bash -b ${GIT_BRANCH}
```

Comment:
Since users will typically use the `build-image.sh` script directly, which calls this `Dockerfile`, do we want to update `build-image.sh` to take a git branch as an argument, and then pass it as a `GIT_BRANCH` arg into `Dockerfile`?

Though, less of a priority since we don't build an Autoware.Auto image (instead, this repo is checked out during the autoware.ai build) for usage in CARMA Platform.